services/src/unit-test/java/uk/gov/moj/sdt/services/cache/ErrorMessagesCacheTest.java

```diff
@@ -171,4 +141,17 @@ void testUncache() {
         verify(mockGenericDao).query(ErrorMessage.class);
         verify(managementMBean).getCacheResetControl();
     }
+
+
+
+    @Test
+    void testSetGenericDao(){
+        SdtManagementMBean mockSdtManagementMBean = mock(SdtManagementMBean.class);
+        ErrorMessageDao mockErrorMessageDao = mock(ErrorMessageDao.class);
+        ErrorMessagesCache errorMessageCache = new ErrorMessagesCache(mockSdtManagementMBean,mockErrorMessageDao);
+        errorMessageCache.setGenericDao(mockGenericDao);
+
+        assertNotNull(errorMessageCache,"Object should have been populated");
+    }
```

Comment:
Rather than setting up a new ErrorMessagesCache object you could use the "cache" instance that's initialised by the setUp() method.  You should then create a new GenericDao object, use the setGenericDao() method to set it and finally use assertEquals() to check that is has been set.  You'll have to use the getAccessibleField() method on the unit test base class to get the field.

Reply:
done